src/NServiceBus.Core/Features/FeatureActivator.cs

```diff
@@ -225,14 +221,24 @@ public FeatureInfo(Feature feature, FeatureDiagnosticData diagnostics)
                 Feature = feature;
             }
 
+            public void AddTaskControllers(IList<FeatureStartupTaskController> controllersToAdd)
+            {
+                foreach (var taskController in controllersToAdd)
+                {
+                    taskControllers.Add(taskController);
+                }
+            }
+
             public FeatureDiagnosticData Diagnostics { get; }
             public Feature Feature { get; }
-            public IList<FeatureStartupTaskController> TaskControllers { get; set; }
+            public IReadOnlyList<FeatureStartupTaskController> TaskControllers => taskControllers;
```

Comment:
What's the benefit of this change? Another option would have been to just assign a new List to the property with the initializer instead of having a backing field and leave the rest as is

Reply:
I had an idea to populate the Diagnostics as well but yours is simpler. I've made the change so please take a look